dask/array/linalg.py

```diff
@@ -829,22 +842,39 @@ def svd(a):
             "Input ndim: {}\n".format(a.shape, a.ndim)
         )
     if nb[0] > 1 and nb[1] > 1:
-        raise ValueError(
+        raise NotImplementedError(
             "Array must be chunked in one dimension only. "
             "This function (svd) only supports tall-and-skinny or short-and-fat "
             "matrices (see da.linalg.svd_compressed for SVD on fully chunked arrays).\n"
             "Input shape: {}\n"
             "Input numblocks: {}\n".format(a.shape, nb)
         )
 
-    # Tall-and-skinny case
-    if nb[0] >= nb[1]:
-        u, s, v = tsqr(a, compute_svd=True)
-        return u, s, v
-    # Short-and-fat case
+    # Single-chunk case
+    if nb[0] == nb[1] == 1:
+        u, s, v = map(asarray, np.linalg.svd(np.asarray(a), full_matrices=False))
```

Comment:
Is the `np.asarray` here to avoid dispatching back to dask's svd?

We might instead consider

```suggestion
        a = a.compute()
        u, s, v = map(asarray, np.linalg.svd(a, full_matrices=False))
```

For the case where we have a Dask array backed by a cupy array on the GPU, this would (I think) keep things on the GPU the whole time.

Reply:
I had originally just done that to avoid dispatch to the same function but tried to make it all lazy instead (as in @mrocklin's comment above).